topics/03_file_handling_and_regex/HW_03_file_handling_and_regex.py

```diff
@@ -0,0 +1,164 @@
+file_paths = ['file1.txt', 'file2.txt','file3.txt']
+
+with open('merged.txt', 'w', encoding="utf-8") as output_file:
+    for file_path in file_paths:
+        with open(file_path, 'r', encoding="utf-8") as input_file:
+            for line in input_file:
+                output_file.write(line)
+# 9 Create a script that reads a configuration file (e.g., config.ini) and prints out the key-value pairs. Use the configparser library to handle the file.
+import configparser
+config = configparser.ConfigParser()
+config.read(file_name)
+for section in config.sections():
+    print(f"[{section}]")
+    for key, value in config.items(section):
+        print(f"{key} = {value}")
+    print()
+ # 10 Write a script that reads a log file (e.g., server.log) and counts how many times a specific error message appears. Print the number of occurrences of that error.
+def count_error_messages(logfile_path, error_prefix):
+    # Initialize a counter to zero
+    error_count = 0
+
+    # Open and read the log file
+    try:
+        with open(logfile_path, 'r') as file:
+            for line in file:
+                # Check if the error prefix exists in the current line
+                if error_prefix in line:
+                    error_count += 1
+    except FileNotFoundError:
+        print(f"Error: The file '{logfile_path}' does not exist.")
+        return
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        return
+
+    # Print the number of times the error message appears
+    print(f"The message '{error_prefix}' appears {error_count} times in the log file.")
+
+    # Specify the path to the log file and the error prefix
+
+
+log_file_path = 'C:/Users/Olena_Smolianchuk/Downloads/server.log'
+error_message_prefix = 'ERROR '
+
+# Call the function
+count_error_messages(log_file_path, error_message_prefix)
+
+if __name__ == "__main__":
+#    read_file()
+    count_error_messages(log_file_path, error_message_prefix)
+
+# REGEX----------------------------------------------------------------------------------
+# 1 Define the regex pattern for validating an email address
+email_regex = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
+# 2 Write a regex to find all words that start with a capital letter.
+import re
+
+# Sample text for demonstration
+text = "Here are Some words that Start With capital Letters and some that do not."
+
+# Regex pattern to match words that start with a capital letter
+pattern = r'\b[A-Z][a-zA-Z]*\b'
+
+# Find all matching words in the text
+capital_words = re.findall(pattern, text)
+
+# Print the list of words
+print(capital_words)
+# 3 Write a regex to match a phone number in the format (123) 456-7890.
+import re
+
+phone_pattern = r'^\(\d{3}\) \d{3}-\d{4}$'
+def is_valid_phone_number(phone):
+    if re.match(phone_pattern, phone):

```

Comment:
This is how this piece of code can look like, using conditional expression:
`return True if re.match(phone_pattern, phone) else False`